### ISSUE TITLE
Codemods

### ISSUE THREAD

**Issue by lamartire** [lamartire/sharec#116]
This is a very big and important feature. We should allow users to add codemods for their files in sharec configuration packages.
May be, it should looks like `inject` in `hygen`. I don't know at this moment, how to do it better, but I guess - this is a really useful stuff.

**Comment by lamartire** [comment]
## Core concept

So, I have two ideas, how to implement this feature.
The first, allow users to ship `codemods` directly in configuration package:
```
configs/
  .eslintrc
  ...
codemods/
  mod1.js
  mod2.js
  ...
package.json
```
If it would be implement this behaviour -- user should not create additional packages and just update `sharec` and add `codemods` files.

**PROS**:
- Friendly UX
- Easy to ship everything you need in your new project

**CONS**:
- May be, it would be hard to manage configs and codemods in one package. Anyway, it can work only with cli flag, like `--mods`

---
The second implementation -- separated utility for packaging only `codemods`. Like `sharec-codemods`:

```
codemodes/
  mod1.js
  mod2.js
  ...
package.json
```
Then, user should add it to target package.

**PROS**:
- Run without flags as separated utility
- Ship codemods separately of main configuration package

**CONS**:
- Unclear execution order on `postinstall` hook with main configuration package

**Comment by lamartire** [comment]
## Codemods files format

Like I said -- I like `hygen` templates format, but I want to have much cleaner frontmatter.

```yaml
--- 
file: string|RegExp 
after: boolean|string|RegExp 
before: boolean|string|RegExp 
--- 
string 
```
Let me explain.

### `file` 

Target filename, or `RegExp`. Also it can contain wildcards.
`webpack.config.js` -- will be applied only to root `webpack.config.js` file
`*.js` -- will be applied for each `js` file in root of the project
`**/*.js` -- will be applied for each `js` file everywhere in the project
`/\S+(js|ts)$/` -- will be applied to `js` or `ts` files everywhere in the project

---

### `after` 
Marker for determining injecting place
`true` -- will be injected at the end of file
`// anchor` -- will be injected after `// anchor` line
`/^\/{2}\s?anchor/` -- the same as before, but as `RegExp`

---

### `before`
Marker too, like `after`
`true` -- will be injected at the top of file
`// anchor` -- will be injected before `// anchor` line
`/^\/{2}\s?anchor/` -- the same as before, but as `RegExp`

**Comment by lamartire** [comment]
As "core" users, @ai, @sergeysova, @SanderV1992 what do you think about this feature?

**Comment by lamartire** [comment]
Will it be useful, as I see it?

**Comment by ai** [comment]
@lamartire sorry, I have no use case for it to have an opinion

**Comment by lamartire** [comment]
@ai I see. If no one will not to have potential use cases too -- I will close this issue and feature too.

**Comment by sergeysova** [comment]
@lamartire maybe integrate already existing solution for codemods?

https://github.com/facebook/jscodeshift For example?

**Comment by lamartire** [comment]
@sergeysova good idea. Do you have any need to use this feature?

**Comment by sergeysova** [comment]
Not now

**Comment by lamartire** [comment]
Ok, I will freeze this feature now.
